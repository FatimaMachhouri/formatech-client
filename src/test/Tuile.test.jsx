import React from 'react'
import ReactDOM from 'react-dom'
import { create } from 'react-test-renderer'
import Tuile from '../components/tuile'
import TuileNav from '../components/tuileNav'

describe("Tuile component", () => {
    test("renders without crashing", () => {
        const div = document.createElement('div')
        ReactDOM.render(<Tuile name='' formationName=''/>, div)
    })
})

describe("TuileNav component", () => {
    test("renders without crashing", () => {
        const div = document.createElement('div')
        ReactDOM.render(<TuileNav formation=''/>, div)
    })
})

describe("Tuile component", () => {
  test("matches the snapshot", () => {
    expect(<Tuile name='' formationName='IG'/>).toMatchSnapshot();
  });
});

describe("TuileNav component", () => {
    test("matches the snapshot", () => {
      expect(<TuileNav formation='IG'/>).toMatchSnapshot();
    });
  });

describe("Tuile component", () => {
    test("is showed as it is visible by default", () => {
      const component = create(<Tuile name="" formationName="" />);
      const instance = component.getInstance();
      expect(instance.state.visibility).toBe("container");
    });
});

describe("Tuile component", () => {
    test("is closed if asked", () => {
      const component = create(<Tuile name="" formationName="" />);
      const instance = component.getInstance();
      instance.close();
      expect(instance.state.visibility).toBe("hidden");
    });
});
