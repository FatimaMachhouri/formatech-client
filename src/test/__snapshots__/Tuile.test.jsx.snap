// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tuile component matches the snapshot 1`] = `
<Tuile
  formationName="IG"
  name=""
/>
`;

exports[`TuileNav component matches the snapshot 1`] = `
<TuileNavigation
  formation="IG"
/>
`;
